### PR TITLE
Further reduce core-modules profile list

### PR DESCRIPTION
**Pull request by aahmed-se**
* Further reduce core-modules profile list
* Filter project by profile in functions.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1799,14 +1799,12 @@ flexible messaging model and an intuitive client API.</description>
...
         <module>pulsar-broker-common</module>
         <module>pulsar-broker</module>
-        <module>pulsar-broker-shaded</module>
         <module>pulsar-client-api</module>
         <module>pulsar-client</module>
         <module>pulsar-client-admin</module>
         <module>pulsar-client-admin-shaded</module>
         <module>pulsar-client-tools</module>
         <module>pulsar-client-tools-test</module>
-        <module>pulsar-client-all</module>
```
Comment (jiazhai) [inline pom.xml:1809]: Why do we need to remove these 2 modules?

Comment (aahmed-se) [reply pom.xml:1809]: We are building a minimal set of modules targeting developers using ide's they can choose the core profile and be more productive.

Comment (jiazhai) [reply pom.xml:1809]: Thanks for the explanation. we may need some docs for this change.

Comment (merlimat) [reply pom.xml:1744]: @jiazhai There's no change for default behavior, this is just to leave option to just do a minimal build build. It's mentioned in README at https://github.com/apache/pulsar#minimal-build-this-skips-most-of-external-connectors-and-tiered-storage-handlers

**Pull request merged**